source/common/stream_info/set_filter_state_object_impl.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "envoy/stream_info/set_filter_state_object.h"
+
+#include "absl/container/flat_hash_set.h"
+
+namespace Envoy {
+namespace StreamInfo {
+
+/**
+ * Implementation of SetFilterStateObject.
+ */
+template <typename T> class SetFilterStateObjectImpl : public SetFilterStateObject<T> {
```

Comment:
I already mentioned this, but I don't think you need the abstract class here. I would just derive from `FilterState::Object ` directly. I'm not sure you even need the accessor functions at all. You could just have a struct with a single set member.